### PR TITLE
podio: remove build_type variant 

### PR DESCRIPTION
**Pull request by vvolkl**
Since this is already verbatim in the base CMakePackage:  https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cmake.py#L94 

It probably helps with maintenance to not duplicate this.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@bernhardkaindl Waiting for your approval to merge this.

**Review by bernhardkaindl — APPROVED** [review]
Looks good to me.

**Pull request merged**
